src/lib.rs

```diff
@@ -53,6 +57,10 @@ pub trait ObjectStore: std::fmt::Display + Send + Sync + Debug + 'static {
     /// Return the bytes that are stored at the specified location.
     async fn get(&self, location: &Path) -> Result<GetResult>;
 
+    /// Return the bytes that are stored at the specified location
+    /// in the given byte range
+    async fn get_range(&self, location: &Path, range: Range<usize>) -> Result<Bytes>;
```

Comment:
What's the expected behavior on out-of-range requests? Do they error or soft-cap at the end of the object? Would probably be nice to add a test for that behavior.